cp: add support for copy filename with ":"

chenhanxiao opened the pull request:
We use ":" as separator for CONTAINER:PATH.
This patch enables copy filename with ":"
to host.

Signed-off-by: Chen Hanxiao chenhanxiao@cn.fujitsu.com

cpuguy83 commented:
The new test failed on Windows.
Can you add `testRequests(SameHostDaemon)` to the top of your test since you're creating dirs that are being used for the test?

duglin commented on integration-cli/docker_cli_cp_test.go:
```diff
@@ -620,3 +620,41 @@ func TestCpToStdout(t *testing.T) {
+	if err := os.Chdir(tmpdir); err != nil {
+		t.Fatal(err)
+	}
+	_, _, err = dockerCmd(t, "cp", cleanedContainerID+":/te:s:t", ".")
```
Would it have been easier to just use `tmpdir` here instead of '.' to avoid all of the chdir stuff?  Not a huge deal, more just wondering if there was a reason for this.

chenhanxiao commented:
@cpuguy83
Sure, I'm not familiar with Windows.
@duglin 
Fine, will do

chenhanxiao committed:
cp: add support for copy filename with ":"

We use ":" as separator CONTAINER:PATH.
This patch enables copy filename with ":"
to host.

Signed-off-by: Chen Hanxiao <chenhanxiao@cn.fujitsu.com>

chenhanxiao commented:
@cpuguy83 @duglin 
updated, please check.

jfrazelle commented:
LGTM

jfrazelle commented:
this probably closes an issue, so that needs to be in the first commet as fix #blah

duglin commented:
LGTM

duglin commented:
ping @cpuguy83 for the merge

chenhanxiao commented:
@jfrazelle 
I made a quick search on open issues with key words "docker cp", found no related issues with this PR.

crosbymichael commented:
LGTM
